MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -1,891 +1,1774 @@
+            // loop thru {1,1}+{1,n} nsharedjets and remove just one shared track from v0 if a |dPhi(v0,one shared track)| > pi/2
+            if (sharedjet_tool.first) {
+              if (verbose)
+                printf("start shj implementation to {1,1} and {1,n} \n");
+              std::vector<std::vector<size_t>> sv_lonesharedtrack_which_trkidx = sharedjet_tool.second;
+              std::vector<size_t> sv0_lonesharedtrack_which_trkidx = sv_lonesharedtrack_which_trkidx[0];
+              std::vector<size_t> sv1_lonesharedtrack_which_trkidx = sv_lonesharedtrack_which_trkidx[1];
+              if (verbose) {
+                printf("size of set of lone shared tracks per sv0: %lu \n", sv0_lonesharedtrack_which_trkidx.size());
+                printf("size of set of lone shared tracks per sv1: %lu \n", sv1_lonesharedtrack_which_trkidx.size());
+              }
+              for (size_t k = 0; k < sv0_lonesharedtrack_which_trkidx.size(); k++) {
+                track_vec tks_sv0 = sv_total_track_which_trk_vec[vtxidx0];
+                size_t idx = sv0_lonesharedtrack_which_trkidx[k];
+                h_resolve_shared_jets_lonetrkvtx_dphi->Fill(fabs(reco::deltaPhi(tks_sv0[idx]->phi(), phi0)));
+                if (fabs(reco::deltaPhi(tks_sv0[idx]->phi(), phi0)) > M_PI / 2) {
+                  sv_total_track_which_trkidx[vtxidx0].erase(std::remove(sv_total_track_which_trkidx[vtxidx0].begin(), sv_total_track_which_trkidx[vtxidx0].end(), idx),
+                      sv_total_track_which_trkidx[vtxidx0].end());
+                }
+              }
+              if (verbose) {
+                printf("sv0'idx: %lu with ntrack before: %u",vtxidx0, sv0.nTracks());
+                printf("sv1'idx: %lu with ntrack before: %u",vtxidx1, sv1.nTracks());
+              }
+
+              track_set  sv0_resolved_sharedtracks_trkset;
+              for (unsigned int i = 0; i < sv_total_track_which_trkidx[vtxidx0].size(); ++i) {
+                int idx = sv_total_track_which_trkidx[vtxidx0][i];
+                track_vec tks_sv0 = sv_total_track_which_trk_vec[vtxidx0];
+                sv0_resolved_sharedtracks_trkset.insert(tks_sv0[idx]);
+
+              }
+              std::vector<reco::TransientTrack> sv0_resolved_sharedtracks_ttks;
+              for (auto tk : sv0_resolved_sharedtracks_trkset)
+                sv0_resolved_sharedtracks_ttks.push_back(tt_builder->build(tk));
+
+              reco::Vertex sv0_resolved_sharedtracks;
+
+              for (const TransientVertex& tv : kv_reco_dropin(sv0_resolved_sharedtracks_ttks))
+                sv0_resolved_sharedtracks = reco::Vertex(tv);
+
+              sv0 = sv0_resolved_sharedtracks;
+              if (verbose) {
+                printf("sv0'idx: %lu with ntrack after: %u",vtxidx0, sv0.nTracks());
+                printf("sv1'idx: %lu with ntrack after: %u",vtxidx1, sv1.nTracks());
+              }
+            }
+          }
+        }
+      }
+    }
+
+    std::set<int> trackicity;

```

Comment:
Unused and can drop, right?

Reply:
I assume yes, and did so in https://github.com/DisplacedVertices/cmssw-usercode/commit/1b5e9bd35e309caa0baa94c94b26e98ef84a70aa. Take a look at the other small changes I made, mainly targeted at some cleanup:
- when we need the index `i`, I think it's usually safer to increment it as part of the loop in case the `++i` code at the end of the loop is ever deleted by mistake--there are other ways to handle it too though
- at least for our purposes, the two `static_cast`s that I deleted seemed to be unneeded, and the compiler didn't throw any sort of warning when removing them. We probably could change `match_track_jet` to take an unsigned int as input anyway, but I think it's okay as-is.

Just let me know if you see any issue with this! Otherwise we can close this comment as well.

Reply:
Yes, you can remove it since this has not been an issue. I included this part, just in case.